accepted/0022-git-buildpack.md

```diff
@@ -0,0 +1,61 @@
+# Git Support
+
+## Proposal
+
+Introduce an (optional) Git buildpack into each language buildpack family. The buildpack
+will be charged with providing `git` metadata as build-time and run-time environment
```

Comment:
I totally get why this is valuable at build time, but why is a runtime environment variable desirable?

Reply:
I can imagine how providing this metadata through runtime environment variables might be useful for applications written in dynamic languages. Compiled languages might let you bake some of these values into the built executable, but unless we write these values down somewhere dynamic language applications won't have access to them at runtime.

Reply:
I am not as familiar with the typical patterns for doing things like this in a language like ruby or node(so let me know if I am way off base here), but I would have imagined that developers would want to template this into a static file or something like that a build time. The reason I would assume this: it seems like a static property of the build, not something that could/should be configured/injected at runtime.

I am instinctively wary of introducing coupling between the buildpack and the actual application code.